extensions/sql-migration/src/api/utils.ts

```diff
@@ -865,3 +865,21 @@ export async function getBlobLastBackupFileNamesValues(lastFileNames: azureResou
 	}
 	return lastFileNamesValues;
 }
+
+export function getCorrectArmEndpointForAccount(account: AzureAccount): string {
+	// Looks up the correct ARM endpoint for the given account, as defined in extensions\azurecore\src\account-provider\providerSettings.ts
+	switch (account.properties.providerSettings.id) {
```

Comment:
these values are already defined in the providerSettings.ts, we should reference them instead of redefining.

Reply:
I did this for the `getCorrectArmEndpointForAccount` implementation in azurecore, but was having some troubles doing the same for sql-migration (I think because providerSettings.ts is not defined in the same extension), but maybe I just missed something. Let me try again

Reply:
Yeah, I can't seem to get this working. I'm trying the exact same implementation as in azurecore, like [here](https://github.com/microsoft/azuredatastudio/pull/20304/files#diff-347d8b77019786cc29710a78e48c961ffa3ee7f3eb01d446f95f332c065d0e5bR516), using the following import for providerSettings:

`import providerSettings from '../../../azurecore/src/account-provider/providerSettings';`

But I can't get it to build properly. Maybe because this is an import from outside of this extension. Any other ideas? @alanrenmsft @aasimkhan30

Reply:
I'm not super familiar with imports in typescript, but that was the only way that seemed like it could have worked. If I just import azurecore then providerSettings doesn't seem to be accessible

Reply:
i see, they are in different extensions. if you really need to know the endpoints here, you need to add it to the API that azurecore extension exposes: https://github.com/microsoft/azuredatastudio/blob/97a4ecdfbb6216c86defdaa1de2d6430b309c3a6/extensions/azurecore/src/azurecore.d.ts#L271

Reply:
Ah, I see. That might be the best option since it'd be nice to have this identical functionality in both azurecore and sql-migration. Let me try that.

Reply:
That seems to have worked, thanks @alanrenmsft. I've added a new function `getProviderMetadataForAccount` (figured it's a little more generic than the original `getCorrectArmEndpointForAccount` I defined) to the azurecore API in my latest iteration.